sdk/python/kfp/dsl/_types.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class MetaType:
+	'''MetaType is a base type for all scalar and artifact types.
+	'''
+	pass
+
+# Primitive Types
+class MetaInt(MetaType):
+	openapi_schema_validator = '''{
+		"type": "integer"
+	}'''
+
+def Integer(attr={}):
+	return type('Integer', (MetaInt, ), attr)
+
+class MetaString(MetaType):
+	openapi_schema_validator = '''{
```

Comment:
Why not have this as a python structure (dict vs string)?

Reply:
Yes, I converted to simple python classes.
Then, these type instances are converted to dict for consistency checking.

Reply:
I meant the JsonSchema.

Reply:
oh, yes. Done.